app/front/scripts/services/package.js

```diff
@@ -81,13 +81,20 @@ angular.module('Application')
               });
             })
             .then(function(fileOrUrl) {
-              var status = ValidationService.validateResource(fileOrUrl);
-              state.status = status;
-
-              return status.$promise.then(function(results) {
-                encoding = results.encoding;
-                return fileOrUrl;
-              });
+              state.status = {
+                state: 'checking'
+              };
+              return ValidationService.validateResource(fileOrUrl)
+                .then(function(report) {
+                  encoding = report.tables[0].encoding;
+                  state.status.state = 'completed';
+                  state.status.report = report;
+                  return fileOrUrl;
+                })
+                .catch(function(error) {
+                  state.status.state = null;
```

Comment:
Is `state.status` modified by anyone else? If so, should it contain data added/modified by different parts of the codebase? If not, I think it's a good defensive practice to change the entire `state.status`, instead of its internal properties. This also makes the intent of the code clearer, as if you're changing just a single property, it makes me wonder what else is in `state.status`.

Reply:
TBH I don't know. I was barely able to understand how it works at all. It seems this `state.status.x` is changing in many places across the codebase. So I've just copied its usage from the former code to make this PR minimal - maximising chances to do not break the whole app.

Reply:
Yeah, this is very tricky. In an abstract level, I don't understand why we would ever want a `status.state` coming from one place, and a `status.report` coming from another. As we're writing `state.status` ourselves, I think we're safe to overwrite it completely here, even if this pattern isn't followed in other parts of the code. At least this start moving us towards safer coding patterns.